### ISSUE TITLE
Clicking login button in unity metamask will not pop up to login/connect on first load

### ISSUE THREAD

**Issue by minutd97** [ChainSafe/web3.unity#119]
Clicking login button in unity metamask will not pop up to login/connect on first load(opening browser), after reloading the page it will work (tested on empty project).

1. Create a new unity project
2. Switch platform to WebGL
3. Import web3.unitypackage
4. Go to Web3Unity/Scenes and add WebLogin scene to build settings
5. Build and run the project (make sure no browser is open)
6. Click on login button
7. Nothing will happen
8. After reloading the page it works

Clicking login button shoud work at first load without reloading the page.

 - OS: Windows 10
 - Browsers : Chrome/Brave/Firefox

**Comment by KBryan** [comment]
Can you paste what you see in the console debug on clicking the button the first time? I'm not seeing it on my end.

**Comment by minutd97** [comment]
i will give you a video

https://user-images.githubusercontent.com/46588765/160842932-fffce58f-c6a5-4dbf-9e47-1a979c69e88b.mp4

**Comment by KBryan** [comment]
From my quick research. This seems to be an issue with the browser, not the SDK itself. Are you making a call in your project that could cause this to be thrown the first time?

**Comment by minutd97** [comment]
Absolutely not, i just imported the unity package, it's just an empty project with de example login scene, this also happens on brave and firefox, tested also on diferent computers.

**Comment by minutd97** [comment]
My unity version is 2020.3.28f1(LTS)

**Comment by minutd97** [comment]
Here is the full video of the issue

https://drive.google.com/file/d/1rb8iC1vJ9NhMrNQMIR8q8Md-YWgrgasd/view?usp=sharing

**Comment by KBryan** [comment]
Okay, I've double retested this and do not see what you are describing. This build has been out for a while and this is the first report of this. I'm wondering if this has something to do with Unity's webserver. In the build folder, you created for this project. Try running the following command. `python3 -m http.server`  and opening your browser to `http://127.0.0.1:8080/` and see if there is any difference.

**Comment by minutd97** [comment]
This also happens on dedicated server, but i can not give you the link, it is confidential.

**Comment by minutd97** [comment]
i tryed (python3 -m http.server and opening your browser to http://127.0.0.1:8080/) your method but still no change, same thing, works only after reload

**Comment by minutd97** [comment]
Here is my build, can you try to run it and see if it's same behaviour ?

[Build.zip](https://github.com/ChainSafe/web3.unity/files/8381289/Build.zip)

**Comment by KBryan** [comment]
I noticed that in the first load you are hitting a 404 and the only other thing I noticed is that you are disabling gzip. Is there a reason you are doing this?  

![Screenshot 2022-03-30 103023](https://user-images.githubusercontent.com/681817/160862853-ec458574-29c5-4a15-96a6-949d11f0c330.png)

**Comment by minutd97** [comment]
I use Brotli on dedicated server, i used disabled to make the build faster and the 404 warnning to be honest idk what is that... any ideea ? In the meantime i installed a newer version of unity to test out

**Comment by minutd97** [comment]
I made the build Unity version 2021.2.17f1 with Brotli/Gzip/Disabled same behaviour, after reloading the page the warning 404 is still there and it works to connect. I just don't get what is going on :(

**Comment by minutd97** [comment]
Even if you open the browser, open a random page and then open the game it will work but closing and opening the broser again and going straight into the game get's this issue

**Comment by minutd97** [comment]
I found an issue on MetaMask related to this one. Here's the link : https://github.com/MetaMask/metamask-extension/issues/14283

**Comment by minutd97** [comment]
i've already done that and the result is the same

**Comment by KBryan** [comment]
Okay, I ran into this issue today also. Although, for me, this happened because I wasn't logged in to my account. If I'm already logged into my MetaMask extension. I don't see this and the extension pops up as normal. Are you seeing the same thing also?

**Comment by KBryan** [comment]
Although, this doesn't happen all the time and seems to be more intermittent. Which is almost worse.

**Comment by KBryan** [comment]
Okay, for me it was an extension that I had installed a couple of days ago. Try to disable all extensions except for MetaMask and see if that works also.

**Comment by minutd97** [comment]
Metamask is the only extension that i have tbh, but this clears the "Unchecked runtime.lastError: Could not establish connection. Receiving end does not exist." error and i don't think that is the problem anyway

**Comment by KBryan** [comment]
This issue hasn't surfaced in the recent updates to the SDK. Closing this issue.
